src/response.rs

```diff
@@ -208,13 +228,17 @@ fn new_gzip(mut res: ::hyper::client::Response) -> Decoder {
         inner: res,
     };
     match gzip::Decoder::new(reader) {
-        Ok(gzip) => Decoder::Gzip {
-            decoder: gzip,
-            head: head,
-        },
-        Err(e) => Decoder::Errored {
-            err: Some(e),
-            head: head,
+        Ok(gzip) => {
+            Decoder::Gzip {
+                decoder: gzip,
+                head: head,
+            }
+        }
```

Comment:
I think this is https://github.com/rust-lang-nursery/fmt-rfcs/issues/61 as well.